src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js

```diff
@@ -211,6 +262,117 @@ export const WithObjectsAPIRegistrationBackends = {
+          objecttype:
+            'https://objecttypen.nl/api/v1/objecttypes/2c77babf-a967-4057-9969-0200320d23f1',
+          objecttypeVersion: 2,
+          variablesMapping: [
+            {
+              variableKey: 'formioComponent',
+              targetPath: ['path', 'to.the', 'target'],
+            },
+            {
+              variableKey: 'userDefined',
+              targetPath: ['other', 'path'],
+            },
+          ],
+        },
+      },
+    ],
+    registrationPluginsVariables: [
+      {
+        pluginIdentifier: 'objects_api',
+        pluginVerboseName: 'Objects API registration',
+      },
+    ],
+    onFieldChange: data => {
+      console.log(data);
+    },
+  },
+  parameters: {
+    msw: {
+      handlers: [
+        mockTargetPathsPost({
+          string: [
+            {
+              targetPath: ['path', 'to.the', 'target'],
+              isRequired: true,
+              jsonSchema: {type: 'string'},
+            },
+            {
+              targetPath: ['path', 'to', 'uri'],
+              isRequired: true,
+              jsonSchema: {
+                type: 'string',
+                format: 'uri',
+              },
+            },
+          ],
+          object: [
+            {
+              targetPath: ['other', 'path'],
+              isRequired: false,
+              jsonSchema: {type: 'object', properties: {a: {type: 'string'}}, required: ['a']},
+            },
+          ],
+          array: [
+            {
+              targetPath: ['path', 'to', 'array'],
+              isRequired: true,
+              jsonSchema: {type: 'array'},
+            },
+          ],
+        }),
+      ],
+    },
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    const editIcons = canvas.getAllByTitle('Registratie-instellingen bewerken');
+
+    // The second icon is for the single file upload component variable
+    userEvent.click(editIcons[1]);
+
+    const targetSchemaDropdown = await screen.findByRole('combobox');
+
+    await expect(targetSchemaDropdown).toBeInTheDocument();
+
+    await waitFor(async () => {
+      // Only the targets of type string should appear
+      await expect(
+        screen.getByRole('option', {name: 'path > to.the > target (required)'})
+      ).toBeVisible();
+      await expect(screen.getByRole('option', {name: 'path > to > uri (required)'})).toBeVisible();
+    });
+
+    const saveButton = screen.getByRole('button', {name: 'Opslaan'});
+    userEvent.click(saveButton);
+
+    // The third icon is for the multiple file upload component variable
+    userEvent.click(editIcons[2]);
+
+    const dropdown = await screen.findByRole('combobox');
+
+    await expect(dropdown).toBeInTheDocument();
+
+    await waitFor(async () => {
+      // Only the targets of type array should appear
+      await expect(
+        screen.getByRole('option', {name: 'path > to > array (required)'})
+      ).toBeVisible();
+    });
```

Comment:
can't you do:

```suggestion
    // Only the targets of type array should appear
    expect(await screen.findByRole('option', {name: 'path > to > array (required)'})).toBeVisible();
```
`findBy*` is basically `await waitFor( getBy* )`

Reply:
I will try again though, maybe I did something else wrong when I tried

Reply:
I had forgot the `await` inside the expect :facepalm: I'm a silly

Reply:
yeah, the await for `find*` is needed, the `await` before `expect` isn't :)

async is hard!